src/api/java/com/minecolonies/api/inventory/container/ContainerCitizenInventory.java

```diff
@@ -78,11 +78,11 @@ public ContainerCitizenInventory(final int windowId, final PlayerInventory inv,
             final ICitizenData data;
             if (citizenId > 0)
             {
-                data = colony.getCitizenManager().getCitizen(citizenId);
+                data = (ICitizenData) colony.getCitizenManager().getCivilian(citizenId);
```

Comment:
This is a bit weird.
Could you check if something with generics works here?

Reply:
true thats weird, hm I think I actually did that generic let me check, would've to cast everywhere else

Reply:
ye there is already generic support for it, just forgot to change the return type, good catch^^